src/core.jl

```diff
@@ -1,13 +1,30 @@
-abstract Detector
-abstract DescriptorParams
+abstract Params
```

Comment:
Changed `DescriptorParams` to `Params` as many algorithms do the work of Detection and Description both, examples being ORB and BRISK.